### PR TITLE
Fix resource leak. Close InetUtils and nested executor service

### PR DESCRIPTION
**Pull request by igor-suhorukov**

### CODE REVIEW

**Comment by spencergibb** [comment]
Can you explain your reasoning? The reason it is assigned is that it gets reused. What resource leak is there?

**Comment by igor-suhorukov** [comment]
@spencergibb it used only as local variable and this reference will be lost after return. This case detected by static code analysis. Under the hood InetUtils open executor service

**Comment by spencergibb** [comment]
`InetUtils` should be a spring bean and be subject to a bean lifecycle (ie: `Closable` beans will get their `close()` method called. In what situation do you encounter the case where `inetUtils` is null?

**Comment by spencergibb** [comment]
It's rather require `InetUtils` to not be null, then remove the `if` statement.

**Comment by igor-suhorukov** [comment]
>> In what situation do you encounter the case where inetUtils is null?
So we just need 
		if (inetUtils == null) {
		    throw new IllegalArgumentException("inetUtils");
		}

**Comment by spencergibb** [comment]
I'd reccomend using Springs `Assert` class in the constructor.
```
		Assert.notNull(inetUtils, "inetUtils may not be null");
		this.firstNonLoopbackAddress = findFirstNonLoopbackAddress(inetUtils);
```

**Comment by igor-suhorukov** [comment]
thanks  @spencergibb  This advice sounds reasonable

**Comment by spencergibb** [comment]
@igor-suhorukov looks like some tests don't pass in an `InetUtils` and failed, please update.

File: spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin2/ZipkinAutoConfiguration.java [hunk 1]
```diff
@@ -93,6 +94,12 @@ public ZipkinRestTemplateCustomizer zipkinRestTemplateCustomizer(ZipkinPropertie
 		return new DefaultZipkinRestTemplateCustomizer(zipkinProperties);
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	InetUtils zipkinInetUtils(){
```
Comment (spencergibb) [inline spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin2/ZipkinAutoConfiguration.java]: I don't think this is needed. Why did you add this?

Comment (igor-suhorukov) [reply spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin2/ZipkinAutoConfiguration.java]: Because you say:
`InetUtils should be a spring bean and be subject to a bean lifecycle (ie: Closable beans will get their close() method called. `

**Comment by spencergibb** [comment]
But it's already created by a different project
